### ISSUE TITLE
Manual approval always causes a new plan to be created

### ISSUE THREAD

**Issue by rhart** [weaveworks/tf-controller#428]
tf-controller is just what I've been waiting for, so much potential for this project!  But I've been experimenting with manual approval and have a problem.  Basically a plan is being generated successfully, but every time I push a commit with the given approvePlan id, it seems to trigger a new reconciliation and a new plan.  If I edit the Terraform resource with the given approvePlan id, it applies the plan as expected.  Any idea what I might be doing wrong here?

Version 0.8.3 of the Helm chart

Approving plan `plan-main-792c7645f0` ends up creating a new plan, `plan-main-845ca323f7`

Controller logs
```
{"level":"info","ts":"2022-10-17T14:57:39.269Z","msg":">> Started Generation: 7","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:39.269Z","msg":"getting source","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:39.269Z","msg":"before lookup runner: checking ready condition","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0","ready":"&Condition{Type:Ready,Status:True,ObservedGeneration:0,LastTransitionTime:2022-10-17 14:55:00 +0000 UTC,Reason:TerraformPlannedWithChanges,Message:Plan generated: set approvePlan: \"plan-main-86b82f3172\" to approve this plan.,}"}
{"level":"info","ts":"2022-10-17T14:57:39.269Z","msg":"before lookup runner: updating status","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0","ready":"&Condition{Type:Ready,Status:True,ObservedGeneration:0,LastTransitionTime:2022-10-17 14:55:00 +0000 UTC,Reason:TerraformPlannedWithChanges,Message:Plan generated: set approvePlan: \"plan-main-86b82f3172\" to approve this plan.,}"}
{"level":"info","ts":"2022-10-17T14:57:39.289Z","msg":"before lookup runner: updated status","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0","ready":"&Condition{Type:Ready,Status:Unknown,ObservedGeneration:0,LastTransitionTime:2022-10-17 14:57:39.269210951 +0000 UTC m=+7714.941036499,Reason:Progressing,Message:Reconciliation in progress,}"}
{"level":"info","ts":"2022-10-17T14:57:39.290Z","msg":"trigger namespace tls secret generation","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:39.290Z","logger":"cert-rotation","msg":"TLS already generated for ","namespace":"flux-system"}
{"level":"info","ts":"2022-10-17T14:57:39.290Z","msg":"show runner pod state: ","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0","name":"tf-test","state":"running"}
{"level":"info","ts":"2022-10-17T14:57:54.313Z","msg":"runner is running","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.330Z","msg":"setting up terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.592Z","msg":"write backend config: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.593Z","msg":"new terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0","workingDir":"/tmp/flux-system-tf-test/apps/gradle-cs-dev/ge-dev-eu-central-1-green/tf-test/plans"}
{"level":"info","ts":"2022-10-17T14:57:54.597Z","msg":"generate vars from tf: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.597Z","msg":"generated var files from spec","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.598Z","msg":"generate template: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:54.598Z","msg":"generated template","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:59.177Z","msg":"init reply: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:59.177Z","msg":"tfexec initialized terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:59.178Z","msg":"workspace select reply: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:57:59.178Z","msg":"calling plan ...","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:04.041Z","msg":"plan: ok, found drift: true","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:06.001Z","msg":"save tfplan: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:06.002Z","msg":"Planned.\nPlan generated: set approvePlan: \"plan-main-792c7645f0\" to approve this plan.","name":"tf-test","namespace":"flux-system","reconciler kind":"Terraform","reason":"Progressing","annotations":{"infra.contrib.fluxcd.io/revision":"main/792c7645f0bd8703a854b0c038ffa1016b79b1b3"}}
{"level":"debug","ts":"2022-10-17T14:58:06.002Z","logger":"events","msg":"Planned.\nPlan generated: set approvePlan: \"plan-main-792c7645f0\" to approve this plan.","type":"Normal","object":{"kind":"Terraform","namespace":"flux-system","name":"tf-test","uid":"603d6d13-9dea-4814-88a4-87f513368d9b","apiVersion":"infra.contrib.fluxcd.io/v1alpha1","resourceVersion":"103333416"},"reason":"Progressing"}
{"level":"info","ts":"2022-10-17T14:58:06.022Z","msg":"should apply == false","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:07.181Z","msg":"clean up dir: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:07.195Z","msg":"Reconciliation completed. Generation: 7","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T14:58:07.195Z","msg":"Reconciliation is stopped to wait for a manual approve","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"7dfe0f3d-fe6e-445d-b3da-02b99fe6a3f0"}
{"level":"info","ts":"2022-10-17T15:04:29.307Z","msg":">> Started Generation: 7","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:29.307Z","msg":"getting source","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:29.307Z","msg":"before lookup runner: checking ready condition","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33","ready":"&Condition{Type:Ready,Status:True,ObservedGeneration:0,LastTransitionTime:2022-10-17 14:57:39 +0000 UTC,Reason:TerraformPlannedWithChanges,Message:Plan generated: set approvePlan: \"plan-main-792c7645f0\" to approve this plan.,}"}
{"level":"info","ts":"2022-10-17T15:04:29.307Z","msg":"before lookup runner: updating status","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33","ready":"&Condition{Type:Ready,Status:True,ObservedGeneration:0,LastTransitionTime:2022-10-17 14:57:39 +0000 UTC,Reason:TerraformPlannedWithChanges,Message:Plan generated: set approvePlan: \"plan-main-792c7645f0\" to approve this plan.,}"}
{"level":"info","ts":"2022-10-17T15:04:29.329Z","msg":"before lookup runner: updated status","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33","ready":"&Condition{Type:Ready,Status:Unknown,ObservedGeneration:0,LastTransitionTime:2022-10-17 15:04:29.307349504 +0000 UTC m=+8124.979175042,Reason:Progressing,Message:Reconciliation in progress,}"}
{"level":"info","ts":"2022-10-17T15:04:29.330Z","msg":"trigger namespace tls secret generation","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:29.330Z","logger":"cert-rotation","msg":"TLS already generated for ","namespace":"flux-system"}
{"level":"info","ts":"2022-10-17T15:04:29.330Z","msg":"show runner pod state: ","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33","name":"tf-test","state":"running"}
{"level":"info","ts":"2022-10-17T15:04:44.353Z","msg":"runner is running","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.369Z","msg":"setting up terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.624Z","msg":"write backend config: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.629Z","msg":"new terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33","workingDir":"/tmp/flux-system-tf-test/apps/gradle-cs-dev/ge-dev-eu-central-1-green/tf-test/plans"}
{"level":"info","ts":"2022-10-17T15:04:44.635Z","msg":"generate vars from tf: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.635Z","msg":"generated var files from spec","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.636Z","msg":"generate template: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:44.636Z","msg":"generated template","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:48.495Z","msg":"init reply: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:48.495Z","msg":"tfexec initialized terraform","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:48.496Z","msg":"workspace select reply: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:48.496Z","msg":"calling plan ...","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:53.126Z","msg":"plan: ok, found drift: true","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:55.222Z","msg":"save tfplan: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:55.222Z","msg":"Planned.\nPlan generated: set approvePlan: \"plan-main-845ca323f7\" to approve this plan.","name":"tf-test","namespace":"flux-system","reconciler kind":"Terraform","reason":"Progressing","annotations":{"infra.contrib.fluxcd.io/revision":"main/845ca323f79540508e7903bc47285a6951d3476b"}}
{"level":"debug","ts":"2022-10-17T15:04:55.223Z","logger":"events","msg":"Planned.\nPlan generated: set approvePlan: \"plan-main-845ca323f7\" to approve this plan.","type":"Normal","object":{"kind":"Terraform","namespace":"flux-system","name":"tf-test","uid":"603d6d13-9dea-4814-88a4-87f513368d9b","apiVersion":"infra.contrib.fluxcd.io/v1alpha1","resourceVersion":"103334475"},"reason":"Progressing"}
{"level":"info","ts":"2022-10-17T15:04:55.243Z","msg":"should apply == false","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:56.382Z","msg":"clean up dir: ok","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:56.406Z","msg":"Reconciliation completed. Generation: 7","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:56.406Z","msg":"Reconciliation is stopped to wait for a manual approve","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"c6a8aa3e-fde0-4f6a-a7de-1aa87cce7f33"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","msg":">> Started Generation: 8","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","msg":"getting source","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","msg":"before lookup runner: checking ready condition","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b","ready":"&Condition{Type:Ready,Status:Unknown,ObservedGeneration:0,LastTransitionTime:2022-10-17 15:04:29 +0000 UTC,Reason:TerraformPlannedWithChanges,Message:Plan generated: set approvePlan: \"plan-main-845ca323f7\" to approve this plan.,}"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","msg":"trigger namespace tls secret generation","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","logger":"cert-rotation","msg":"TLS already generated for ","namespace":"flux-system"}
{"level":"info","ts":"2022-10-17T15:04:56.407Z","msg":"show runner pod state: ","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b","name":"tf-test","state":"running"}
{"level":"info","ts":"2022-10-17T15:05:11.426Z","msg":"runner is running","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b"}
{"level":"info","ts":"2022-10-17T15:05:11.426Z","msg":"reconciliation is stopped to wait for a manual approve","controller":"terraform","controllerGroup":"infra.contrib.fluxcd.io","controllerKind":"Terraform","Terraform":{"name":"tf-test","namespace":"flux-system"},"namespace":"flux-system","name":"tf-test","reconcileID":"86112958-6902-4025-b0c2-5943372d8a8b"}
```

**Comment by chanwit** [comment]
Thank you so much for reporting this @rhart 

Yes, it's a regression caused by introducing many new features during v0.13.0-rc.
I'm addressing it.

**Comment by rhart** [comment]
👍 thanks for the quick response, @chanwit

**Comment by chanwit** [comment]
It is fixed now by #438 and #441 (v0.13.0)

Please feel free to re-open this issue if you still see the problem.

**Comment by rhart** [comment]
Upgraded and works great! Thanks for the quick turnaround.
